torch/nn/parallel/distributed.py

```diff
@@ -665,6 +665,16 @@ def _sync_params_and_buffers(self, authoritative_rank=0):
                 module_states, self.broadcast_bucket_size, authoritative_rank
             )
 
+        # After syncing, tag them as replicated.
+        from torch.distributed._shard.replicated_tensor import ReplicatedTensor, ReplicatedParameter
```

Comment:
Shouldn't we check if each `param` or `buffer` is replicated before converting them?
Similarly, shouldn't we record which ones we wrapped and use that when unwrapping, instead of relying on `parameters_to_ignore`.

Another thing, is using `parameters_to_ignore` to filter buffers just a naming artifact?

Reply:
> Similarly, shouldn't we record which ones we wrapped and use that when unwrapping, instead of relying on parameters_to_ignore.

So `parameters_to_ignore` is slightly orthogonal to tagging something as replicated or not since that option is actually controlling which parameters DDP ignores completely. That is why we always need to rely on it and ensure we don't tag `parameters_to_ignore` as Replicated at all.

> Another thing, is using parameters_to_ignore to filter buffers just a naming artifact?

It's an unfortunate name since it is filtering buffers as well in other places. I'll probably fix this name in a follow up PR.

Reply:
Oops, I added `self.parameters_to_ignore` a couple of years ago and agree that a better name is `parameters_and_buffers_to_ignore`.